spark-jobs/src/main/scala/filodb/repair/ChunkCopierMain.scala

```diff
@@ -0,0 +1,90 @@
+import monix.execution.Scheduler
+import org.apache.spark.SparkConf
+import org.apache.spark.sql.SparkSession
+
+import filodb.cassandra.FiloSessionProvider
+import filodb.cassandra.columnstore.CassandraColumnStore
+import filodb.core.DatasetRef
+
+object ChunkCopierMain extends App {
+  new ChunkCopier().run(new SparkConf(loadDefaults = true))
+}
+
+class ChunkCopier extends StrictLogging {
+
+  // scalastyle:off method.length
+  def run(conf: SparkConf): Unit = {
+
+    val spark = SparkSession.builder()
+      .appName("FiloDBChunkCopier")
+      .config(conf)
+      .getOrCreate()
+
+    logger.info(s"ChunkCopier Spark Job Properties: ${conf.toDebugString}")
+
+    def openConfig(str: String) = ConfigFactory.parseFile(new File(conf.get(str))).getConfig("filodb")
+
+    // Both "source" and "target" refer to file paths which define config files that have a
+    // top-level "filodb" section and a "cassandra" subsection.
+    val sourceConfig = openConfig("spark.filodb.chunkcopier.source.configFile")
+    val targetConfig = openConfig("spark.filodb.chunkcopier.target.configFile")
+
+    val sourceCassConfig = sourceConfig.getConfig("cassandra")
+    val targetCassConfig = targetConfig.getConfig("cassandra")
+
+    val sourceDatasetRef = DatasetRef.fromDotString(conf.get("spark.filodb.chunkcopier.source.dataset"))
+    val targetDatasetRef = DatasetRef.fromDotString(conf.get("spark.filodb.chunkcopier.target.dataset"))
+
+    // Examples: 2019-10-20T12:34:56Z  or  2019-10-20T12:34:56-08:00
+    def parseDateTime(str: String) = Instant.from(DateTimeFormatter.ISO_OFFSET_DATE_TIME.parse(str))
+
+    val ingestionTimeStart = parseDateTime(conf.get("spark.filodb.chunkcopier.ingestionTimeStart"))
+    val ingestionTimeEnd = parseDateTime(conf.get("spark.filodb.chunkcopier.ingestionTimeEnd"))
+
+    val diskTimeToLiveSeconds = conf.getTimeAsSeconds("spark.filodb.chunkcopier.diskTimeToLive")
+    val splitsPerNode = conf.getInt("spark.filodb.chunkcopier.splitsPerNode", 1)
+    val batchSize = conf.getInt("spark.filodb.chunkcopier.batchSize", 10000)
+
+    val readSched = Scheduler.io("cass-read-sched")
+    val writeSched = Scheduler.io("cass-write-sched")
+
+    val sourceSession = FiloSessionProvider.openSession(sourceCassConfig)
+    val targetSession = FiloSessionProvider.openSession(targetCassConfig)
+
+    val sourceCassandraColStore = new CassandraColumnStore(sourceConfig, readSched, sourceSession)(writeSched)
+    val targetCassandraColStore = new CassandraColumnStore(targetConfig, readSched, targetSession)(writeSched)
+
+    val splits = sourceCassandraColStore.getScanSplits(sourceDatasetRef, splitsPerNode)
+
+    logger.info(s"Cassandra split size: ${splits.size}. We will have this many spark partitions. " +
+      s"Tune splitsPerNode which was $splitsPerNode if parallelism is low")
+
+    spark.sparkContext
+      .makeRDD(splits)
+      .foreachPartition{ splitIter =>
+        sourceCassandraColStore.copyChunksByIngestionTimeRange(
```

Comment:
Dont we need to delete the chunks first ?

Reply:
The assumption is that both the source and target are generating chunks with the same time stamp keys. Cassandra inserts and updates are the same and overwrite any existing record. Also, by not deleting the existing chunks, the copy tool can be run both directions safely, and at any time.